packages/mui-base/src/Portal/Portal.tsx

```diff
@@ -33,8 +33,11 @@ const Portal = React.forwardRef(function Portal(
 ) {
   const { children, container, disablePortal = false } = props;
   const [mountNode, setMountNode] = React.useState<ReturnType<typeof getContainer>>(null);
-  // @ts-expect-error TODO upstream fix
-  const handleRef = useForkRef(React.isValidElement(children) ? children.ref : null, forwardedRef);
+  const handleRef = useForkRef(
+    // @ts-expect-error TODO upstream fix
+    React.isValidElement(children) ? (children.props.ref ?? (children as any).ref) : null,
```

Comment:
We should use `getChildRef` from @mui/material once it's moved to a common package.

Reply:
https://github.com/mui/material-ui/pull/43132